ext/nnstreamer/tensor_filter/tensor_filter_edgetpu.cc

```diff
@@ -62,14 +59,237 @@ typedef struct {
       /**< EdgeTPU Device context */
   std::unique_ptr<tflite::FlatBufferModel> model;
       /**< Loaded TF Lite model (from model_path) */
-} pdata;
+  static std::unique_ptr<tflite::Interpreter>
+      BuildEdgeTpuInterpreter(const tflite::FlatBufferModel &model,
+          edgetpu::EdgeTpuContext* edgetpu_context);
+
+
+  /** Internal Utility Functions & Properties ***************************/
+  void cleanup ();
+  static void setTensorProp (tflite::Interpreter *interpreter,
+      const std::vector<int> &tensor_idx_list, GstTensorsInfo & tensorMeta);
+  static int getTensorDim ( tflite::Interpreter *interpreter, int tensor_idx,
+      tensor_dim dim);
+  static tensor_type getTensorType (TfLiteType tfType);
+  static const char *name;
+  static const accl_hw hw_list[];
+  static const int num_hw = 1;
+  static edgetpu_subplugin *registeredRepresentation;
+
+public:
+  static void init_filter_edgetpu ();
+  static void fini_filter_edgetpu ();
+
+  edgetpu_subplugin ();
+  ~edgetpu_subplugin ();
+
+  tensor_filter_subplugin & getEmptyInstance();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+};
+
+const char *edgetpu_subplugin::name = "edgetpu";
+const accl_hw edgetpu_subplugin::hw_list[] = { ACCL_NPU_EDGE_TPU };
```

Comment:
how about also adding DEFAULT and AUTO (parser goes to DEFAULT if no accel is passed as input, and goes to AUTO if the passed input string is unmatched).

Reply:
I suppose that a subpluigin will simply provide a list of hardware accelerators that it supports and a infra (tensor_filter main) should choose appropraitely.

For example if "default or auto" is chosen, a tensor_filter should be able to choose "ACCL_NPU_EDGE_TPU" if the chosen subpluigin is edgetpu. I thought the current tensor-filter does this already. Doesn't it do so?

Reply:
> I suppose that a subpluigin will simply provide a list of hardware accelerators that it supports and a infra (tensor_filter main) should choose appropraitely.

Yes, this exists. However, that does not handle default or auto.

> For example if "default or auto" is chosen, a tensor_filter should be able to choose "ACCL_NPU_EDGE_TPU" if the chosen subpluigin is edgetpu. I thought the current tensor-filter does this already. Doesn't it do so?

The current tensor-filter sets the accelerator backend out of the list provided by the sub-plugin. Choosing which backend to use in case of default or auto is not yet done in tensor_filter main (tensor_filter has no means to decide the default or auto backend for now especially with many filters supporting multiple backends).

Reply:
This issue is registered at https://github.com/nnstreamer/nnstreamer/issues/2320

I'll add AUTO/DEFAULT from edgetpu subplugin as a workaround, which can  be reverted if #2320 is resolved.

Reply:
Added a commit to resolve it.